egg/core/trainers.py

```diff
@@ -167,17 +157,17 @@ def train(self, n_epochs):
             for callback in self.callbacks:
                 callback.on_epoch_begin()
 
-            train_loss, train_rest = self.train_epoch()
+            train_loss, train_interaction = self.train_epoch()
 
             for callback in self.callbacks:
-                callback.on_epoch_end(train_loss, train_rest)
+                callback.on_epoch_end(train_loss, train_interaction)
```

Comment:
train_interaction should be the sum (as implemented in the __add__ method) of many interactions which should be a single interaction

writing this to explain above comments

Reply:
i believe it's fixed already!